ast/desugar/Desugar.cc

```diff
@@ -704,6 +704,12 @@ ExpressionPtr node2TreeImpl(DesugarContext dctx, unique_ptr<parser::Node> what)
         if (what.get() == nullptr) {
             return MK::EmptyTree();
         }
+
+        if (auto expr = what->getCachedDesugaredExpr()) {
```

Comment:
The translations we implement in the `Translator` will fill this in, which we can use here to short-circuit the logic below.

Thinking goes that once we've implemented it for all nodes, none of the code below will be used anymore, and we can just delete and inline all this stuff.